terms-generators.lua

```diff
@@ -1159,17 +1202,22 @@ local function array_unchecked_new_fn(self, array, n)
 	}, self)
 end
 
-local function array_new_fn(self, array, n)
+local function array_new_fn(self, array, first, last)
```

Comment:
I find this change highly questionable given that this is only ever used once, and it simply calls `:new(1, n)`. Why make it more complicated if nobody ever uses it?

Reply:
We have to iterate over `array` anyways in `array_new_fn()`, so I'd prefer to expose it since generally callers of `:new()` (and especially `:new_unchecked()`) care about performance more.

Reply:
That didn't answer my question at all. This is only ever used in one place and it just sets the first argument to 1.

Reply:
More places should be using it, and I have a few drafts that increase usage, but I've had other priorities so far. Other places that would consume `:new()` so far have been managed with `:unchecked_new()`. FWIW, taking `first` and `last` also aligns `:new()` with `:copy()`, which I find satisfying.

Reply:
Also, if `:new()` is going to stick around with a `n`/`last` parameter, which it really needs to when handling arrays with `nil`s, then it should have a `first` parameter and that's what I wish I had originally written for it.